src/natcap/invest/ui_server.py

```diff
@@ -102,6 +104,60 @@ def get_invest_validate():
     return json.dumps(results)
 
 
+@app.route('/colnames', methods=['POST'])
+def get_vector_colnames():
+    """Get a list of column names from a vector.
+    This is used to fill in dropdown menu options in a couple models.
+
+    Body (JSON string):
+        vector_path (string): path to a vector file
+        
+    Returns:
+        a JSON string.
+    """
+    payload = request.get_json()
+    LOGGER.debug(payload)
+    vector_path = payload['vector_path']
+    colnames = []  
+    # a lot of times the path will be empty so don't even try to open it
+    if vector_path:  
+        try:
+            vector = gdal.OpenEx(vector_path, gdal.OF_VECTOR)
+            colnames = [defn.GetName() for defn in vector.GetLayer().schema]
+        except:
+            LOGGER.error(f'Could not read column names from {vector_path}')
+    LOGGER.debug(colnames)
+    return json.dumps(colnames)
+
+
+@app.route('/vector_has_points', methods=['POST'])
+def get_vector_has_points():
+    """Return boolean indicating if a vector may contain points.
+    This is used by the DelineateIt UI to determine if the 'snap points' 
+    option should be enabled.
+
+    Body (JSON string):
+        vector_path (string): path to a vector file
+
+    Returns:
+        a boolean.
+    """
+    payload = request.get_json()
+    LOGGER.debug(payload)
+    vector_path = payload['vector_path']
+    # default True because it may have point geometries unless proven otherwise
+    has_points = True
+    # a lot of times the path will be empty so don't even try to open it
+    if vector_path:
+        try:
+            has_points = delineateit._vector_may_contain_points(vector_path)
+        except:
+            LOGGER.error(
+                f'Could not tell if vector {vector_path} contains points.')
+    LOGGER.debug({'has_points': has_points})
+    return json.dumps({'has_points': has_points})
```

Comment:
I was wondering if the plain boolean value could be returned as valid json without the object syntax, and I learned the answer is no!

```
>>> json.dumps(True)
'true'
>>> json.dumps({'has_points': True})
'{"has_points": true}'
```

Reply:
i wish it didn't have to have an arbitrary key name but no way around it :/

Reply:
Yep, that's unfortunately the case ... conformant JSON objects must have key-value pairs.